lib/ASRouter.jsm

```diff
@@ -104,6 +105,14 @@ const MessageLoaderUtils = {
         .map(msg => ({...msg, provider: provider.id}));
     const lastUpdated = Date.now();
     return {messages, lastUpdated};
+  },
+
+  async installAddonFromURL(browser, url) {
+    const aUri = Services.io.newURI(url);
```

Comment:
Services.io.newURI might throw if the url is invalid, so let's wrap this in a try/catch to hand that, and maybe just send something to telemetry or return out of the function

Reply:
Wrapping this in a try catch and will follow up with telemetry in another bug. Currently the router doesn't have access to the telemetry feed.